apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -46,10 +39,35 @@ public static Task<SpannerClient> AcquireClientAsync()
             s_clientEntryPool.Clear();
         }
 
-        private struct ClientPoolKey
+        private struct ClientPoolKey : IEquatable<ClientPoolKey>
         {
-            public ITokenAccess Credential;
-            public ServiceEndpoint Endpoint;
+            public ITokenAccess Credential { get; }
+            public ServiceEndpoint Endpoint { get; }
+
+            public ClientPoolKey(ITokenAccess tokenAccess, ServiceEndpoint serviceEndpoint) : this()
```

Comment:
Have just thought: exposing `ITokenAccess` here ties us to Google.Apis.Auth, which we're trying to get away from.
It's fine for now as we're only pre-release, but we'll need to think about timing between this and the rest of my "split from Google.Apis.Auth" work. See doc shared in team meeting for more details :)